core/modules/dashboard/dashboard.module

```diff
@@ -0,0 +1,444 @@
+  if (isset($_POST['form_id']) && $_POST['form_id'] == 'user_pass_reset') {
+    return;
+  }
+
+  // Don't redirect if no permission to access dashboard.
+  if (!user_access('access dashboard', $account)) {
+    return;
+  }
+
+  // Don't redirect if redirect setting is not set.
+  if (config_get('dashboard.settings', 'dashboard_login_redirect') != 1) {
+    return;
+  }
+
+  // Check access to layout and redirect.
+  $dashboard = layout_load('dashboard');
+  if ($dashboard && !$dashboard->disabled && $dashboard->checkAccess()) {
+    $_GET['destination'] = 'admin/dashboard';
+  }
+}
+
+/**
+ * Implements hook_preprocess_page().
+ */
+function dashboard_preprocess_page(&$variables) {
+  if (arg(0) == 'admin' && arg(1) == 'dashboard') {
+    // Include dashboard specific style sheet.
+    $css_path = backdrop_get_path('module', 'dashboard') . '/css/dashboard.css';
+    backdrop_add_css($css_path);
+    $variables['classes'][] = 'dashboard';
+  }
+}
+
+/**
+ * Implements hook_preprocess_block().
+ */
+function dashboard_preprocess_block(&$variables) {
+  if ($variables['layout']->name == 'dashboard'
+     && $variables['block']->module == 'dashboard') {
+    // Add dashboard specific block class.
+    $variables['classes'][] = 'admin-panel';
+  }
+}
+
+/**
+ * Implements hook_theme()
+ */
+function dashboard_theme() {
+  $base = array(
+    'file' => 'dashboard.theme.inc',
+  );
+
+  $items = array(
+    'dashboard_create' => array(
+      'variables' => array('create' => array()),
+    ) + $base,
+    'dashboard_overview_content' => array(
+      'variables' => array('overview' => array()),
+    ) + $base,
+    'dashboard_overview_user' => array(
+      'variables' => array('overview' => array()),
+    ) + $base,
+    'dashboard_admin_table' => array(
+      'variables' => array('header' => array(), 'rows' => array(), 'link' => FALSE),
+    ) + $base,
+  );
+
+  return $items;
+}
+
+/**
+ * Retrieves overview data for all content on the site.
+ *
+ * @param $settings
+ *   Block settings data.
+ *
+ * @return $items
+ *   An array of items for a bulleted list.
+ *
+ */
+function dashboard_get_content_overview($settings = array()) {
+  // Not cached since comment and node CRUD operations are likely to be more
+  // frequent than Dashboard access events.
+  $items = array();
+  $types = node_type_get_types();
+  $comments_exist = module_exists('comment');
+  foreach ($types as $type => $object) {
+    // Compare against type option on block settings.
+    if ((!array_key_exists($type, $settings['types'])) || (isset($settings['types']) && $settings['types'][$type] == $type)) {
+      $type_count = db_query("SELECT count(*) FROM {node} WHERE type = :type and status = 1", array(':type' => $type))->fetchField();
+      $content_data[$type] = format_plural($type_count, '1 ' . $object->name . ' item', '@count ' . $object->name . ' items');
+      // Check if comments module is enabled.
+      if ($comments_exist) {
```

Comment:
We are not using this variable elsewhere, so why not `module_exists('comment')` directly.

Reply:
Because this is inside of a loop, so rather than calling `module_exists()` once for each content type, it's called only once. Seems good to me.